### ISSUE TITLE
amplify push stuck with uploading

### ISSUE THREAD

**Issue by naveenkoduri** [aws-amplify/amplify-cli#7543]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

12.18.4

### Amplify CLI Version

5.0.0

### What operating system are you using?

Windows 10

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

**amplify push** gets stuck with Uploading. This happened multiple times in different scenarios listed below. In essence, whenever i was trying to make a significant change to my GrapQL API. Minor updates to Schema do not have this issue

- When adding a new @key to an existing api
- When adding @auth rule to an existing API 
- When updating @key to an existing api

The bad part of the issue is - whenever this happens, I can't go to AWS Web console as well to check what is going on. I have no clue what went wrong and why it was stuck. Restarting my windows laptop is the only way I can recover from this problem. Then, I have to do **amplify delete**, and create api from scratch.

This did not start with 5.0.0 CLI version. I experienced this behavior with couple of 4.* versions as well

I also made sure I have the settings listed here even though I don't get any error that was mentioend. https://docs.amplify.aws/cli/graphql-transformer/key#deploying-multiple-secondary-indices-gsi

### Expected behavior

- **amplify push** should not get stuck and should provide error messages if any on the CLI
- Appreciate any pointers to help me understand why access to AWS web console is blocked as if I lost internet connection.

### Reproduction steps

- When adding a new @key to an existing api
- When adding @auth rule to an existing api
- When updating @key to an existing api

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
type Msg
@model(
  queries: { get: "getMsg" })
@auth(
    rules: [
      { allow: private, provider: iam, operations: [read, update] } 
    ])
@key(
  name: "idx1",
  fields: ["recepId", "createdAt"],
  queryField: "listMessagesByRecep"
)
@key(
  name: "idx2",
  fields: ["senderId", "createdAt"],
  queryField: "listMessagesBySender"
) 
@key(
  name: "idx3",
  fields: ["dataId", "createdAt"],
  queryField: "listMessagesById"
) {
  id: ID!
  createdAt: String!  
  senderId: String!
  recepId: String!
  msg: String!
  fooId: String!
}


type Foo
@model(queries: { get: "getFoo"})
@auth(
    rules: [
     
      { allow: private, provider: iam, operations: [read, update] } 
    ])
@key(
  name: "gsi1",
  fields: ["recepId", "dataType"],
  queryField: "listFooByrecepId"
)
@key(
  name: "gsi2",
  fields: ["senderId", "dataType"],
  queryField: "listFooBysenderId"
) 
 {
  id: ID!
  dataType: DataType
  recepId: String
  senderId: String
}








```

</details>

### Log output

<details>

```
# Put your logs below this line
2021-06-18T17:18:39.517Z\|info : amplify add api
2021-06-18T17:18:39.565Z\|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-06-18T17:18:39.927Z\|error : amplify-cli-core.banner-message/index.ts.fetch banner message failed
FetchError: request to https://aws-amplify.github.io/amplify-cli/banner-message.json failed, reason: unable to get local issuer certificate
2021-06-18T17:20:05.576Z\|info : amplify import auth
2021-06-18T17:20:05.625Z\|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-06-18T17:20:05.863Z\|error : amplify-cli-core.banner-message/index.ts.fetch banner message failed
FetchError: request to https://aws-amplify.github.io/amplify-cli/banner-message.json failed, reason: unable to get local issuer certificate

**last messages is upload to S3**

2021-06-18T17:25:53.371Z\|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ify-[***]ync-[***]es/[***]308a32e4c9d039ca/[***]cks/[***]ta.json","Bucket":"[***]ify-[***]ystc-[***]ify-[***]437-[***]ment"}])
2021-06-18T17:25:53.372Z\|info : amplify-provider-awscloudformation.upload-appsync-files.uploadAppSyncFiles.upload.s3Client.uploadFile([{"Key":"[***]cks/[***]ces.json"}])
2021-06-18T17:25:53.372Z\|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ify-[***]ync-[***]es/[***]308a32e4c9d039ca/[***]cks/[***]ces.json","Bucket":"[***]ify-[***]ystc-[***]ify-[***]437-[***]ment"}])
2021-06-18T17:25:53.373Z\|info : amplify-provider-awscloudformation.upload-appsync-files.uploadAppSyncFiles.upload.s3Client.uploadFile([{"Key":"[***]cks/[***]ge.json"}])
2021-06-18T17:25:53.374Z\|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ify-[***]ync-[***]es/[***]308a32e4c9d039ca/[***]cks/[***]ge.json","Bucket":"[***]ify-[***]ystc-[***]ify-[***]437-[***]ment"}])
```

</details>

**Comment by attilah** [comment]
Hi @naveenkoduri 

We have fixed a push lockup issue in 5.0.1 that has happened during error scenarios which is something you are running into it seems. If you see the log there are network error related log entries like: 
```
FetchError: request to https://aws-amplify.github.io/amplify-cli/banner-message.json failed, reason: unable to get local issuer certificate
```

By any chance are you behind a proxy? Is it possible that the proxy has a self-signed (not trusted) certificate issues that could lead to blocking HTTPS operations?

**Comment by naveenkoduri** [comment]
ok. I might be behind proxy on my work laptop, but, not sure why push works sometimes. I pulled my app on my mac and it was failing with error saying authRoleName is not passed to the stack. Looking at generated stacks, it appears the AuthRoleName in root stack is not passed as parameter to the child stack. I tried multiple thinks with no luck like updating parameters.json, etc

I ended up creating the Cognito pool with amplify instead of importing existing pool, then push worked. FYI, I had and needed multiple auth rules. Cognito, OIDC, IAM
